pkg/virt-launcher/virtwrap/network/dhcpv6/dhcpv6.go

```diff
@@ -51,17 +54,38 @@ func SingleClientDHCPv6Server(
 }
 
 func (h *DHCPv6Handler) ServeDHCPv6(conn net.PacketConn, peer net.Addr, m dhcpv6.DHCPv6) {
-	log.Log.V(4).Info("Serving a new request")
+	log.Log.V(4).Info("DHCPv6 serving a new request")
 
 	// TODO how can we make sure the request is from the vm? Is filtering requests arrived to the bridge interface enough?
 
 	msg := m.(*dhcpv6.Message)
-	adv, err := dhcpv6.NewAdvertiseFromSolicit(msg)
+
+	var response *dhcpv6.Message
+
+	optIAAddress := dhcpv6.OptIAAddress{IPv6Addr: h.clientIP, PreferredLifetime: h.leaseDuration, ValidLifetime: h.leaseDuration}
+
+	iface, err := net.InterfaceByName(h.serverIface)
+	if err != nil {
+		log.Log.V(4).Info("DHCPv6 - couldn't get the server interface")
+		return
+	}
+	duid := dhcpv6.Duid{Type: dhcpv6.DUID_LL, HwType: iana.HWTypeEthernet, LinkLayerAddr: iface.HardwareAddr}
```

Comment:
I'm guessing this is the Server Identifier Option (and not the client one), right?
If this is so, using `dhcpv6.DUID_LL` may not work well for migrations as the server interface (/bridge) is not preserving its link-layer address (mac).

If the client is suppose to identify its server using the DUID, it will probably fail to do so.
I am just unsure if this is enforced, and if it is not, is this intentional or a bug that will be fixed in the future.

Reply:
I got this answer from an NM developer:

> I think the client should recover quickly from a server-id change, by asking again for a lease. Of course, it would be nice to test this in practice

Reply:
I've added dhcpv6 to the test checking connectivity after migration and the connectivity survives.
Since the lease is infinite there should be no dhcp request after the migration.
In case of a vm restart, the ipv6 information will be lost anyway and a new dhcpv6 solicit request should be sent.
We need to add connectivity tests after restart + restart after migration. From my manual check seems the dhcpv4 stops wokriking after a vm restart.

Anyway, if that's ok by you, I will keep the duid as is, with the mac address. Ok?

Reply:
Sounds good to me, thanks.